tests/cluster/cluster.go

```diff
@@ -0,0 +1,181 @@
+func RemoveTestNamespace() error {
+	By(fmt.Sprintf("Waiting for namespace %s to be removed, this can take a while ...\n", TestNamespace))
+	err := Clientset.CoreV1().Namespaces().Delete(context.TODO(), TestNamespace, metav1.DeleteOptions{})
+	if err != nil {
+		return errors.Wrap(err, "failed to delete test namespace")
+	}
+
+	EventuallyWithOffset(1, func() error {
+		_, err := Clientset.CoreV1().Namespaces().Get(context.TODO(), TestNamespace, metav1.GetOptions{})
+		return err
+	}, 120*time.Second, 5*time.Second).Should(SatisfyAll(HaveOccurred(), WithTransform(apierrors.IsNotFound, BeTrue())), "Should succeed terminating the namespace")
+
+	return nil
+}
+
+func CreatePrivilegedPodWithIp(podName, nadName, cidr string) error {
+	By(fmt.Sprintf("Creating pod %s with priviliged premission and ip %s", podName, cidr))
+	privileged := true
+	podObject := &corev1.Pod{ObjectMeta: metav1.ObjectMeta{
+		Name: podName,
+		Namespace: TestNamespace,
+		// This annotation makes sure the pod is assigned to a node that has this ovs bridge resource
+		Annotations: map[string]string{"k8s.v1.cni.cncf.io/networks": nadName},
+	},
+	Spec: corev1.PodSpec{ Containers: []corev1.Container{{Name: "test",
+		Image:   "alpine",
+		Command: []string{"sh", "-c", fmt.Sprintf("ip address add %s dev net1; sleep 99999", cidr)},
+		SecurityContext: &corev1.SecurityContext{Privileged: &privileged}}}}}
+
+	_, err := Clientset.CoreV1().Pods(TestNamespace).Create(context.TODO(), podObject, metav1.CreateOptions{})
+	if err != nil {
+		return errors.Wrap(err, "failed to wait for pod container to be in Running state")
+	}
+
+	By("waiting for pod container to be in Running state")
+	Eventually(func() bool {
+		pod, err := Clientset.CoreV1().Pods(TestNamespace).Get(context.TODO(), podName, metav1.GetOptions{})
+		if err != nil {
+			return false
+		}
+
+		if len(pod.Status.ContainerStatuses) > 0 {
+			for _, containerStatus := range pod.Status.ContainerStatuses {
+				if containerStatus.Ready != true {
+					return false
+				}
+			}
+
+			return true
+		}
+		return false
+
+	}, 120*time.Second, 5*time.Second).Should(BeTrue(), "should succeed getting pod container to Ready state")
+	return nil
+}
+
+func DeletePodsInTestNamespace() error {
+	By(fmt.Sprintf("cleaning Pods in %s namespace", TestNamespace))
+	podList, err := Clientset.CoreV1().Pods(TestNamespace).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return errors.Wrapf(err, "failed to get pod list in namespace %s", TestNamespace)
+	}
+
+	for _, pod := range podList.Items {
+		err = Clientset.CoreV1().Pods(TestNamespace).Delete(context.TODO(), pod.Name, metav1.DeleteOptions{})
+		if err != nil {
+			return errors.Wrapf(err, "failed to delete pod %s", pod.Name)
+		}
+	}
+
+	Eventually(func() int {
+		podsList, err := Clientset.CoreV1().Pods(TestNamespace).List(context.TODO(), metav1.ListOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		return len(podsList.Items)
+	}, 6*time.Minute, 1*time.Second).Should(BeZero())
```

Comment:
No need for `1*` if it is just a second.

Reply:
Done